scripts/bench/stt-bench.py

```
@@ -0,0 +1,1602 @@
+    """Score pairs from a JSON file (or stdin with "-") and print JSON."""
+    label = "<stdin>" if source == "-" else source
+    raw = sys.stdin.read() if source == "-" else Path(source).expanduser().read_text(encoding="utf-8")
+    payload = load_json_document(raw, label)
+    pairs = payload if isinstance(payload, list) else [payload]
+    scored = []
+    for index, pair in enumerate(pairs):
+        # A bare string in the list would otherwise index by character and
+        # score whatever came out, which is a wrong number rather than an
+        # error, and a missing key would surface as a bare KeyError.
+        if not isinstance(pair, dict):
+            raise SystemExit(f"{label}: pair {index} is not an object: {describe_entry(pair)}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)